app/src/main/java/org/mozilla/fenix/compose/TabImage.kt

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.icons.compose.Loader
+import mozilla.components.browser.icons.compose.Placeholder
+import mozilla.components.browser.icons.compose.WithIcon
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.concept.base.images.ImageLoadRequest
+import mozilla.components.ui.colors.PhotonColors
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A recent tab image.
+ *
+ * @param tab [TabSessionState] that was recently viewed.
+ * @param modifier [Modifier] used to draw the image content.
+ * @param contentScale [ContentScale] used to draw image content.
+ * @param alignment [Alignment] used to draw the image content.
+ * is null.
+ */
+@Composable
+@Suppress("LongParameterList")
+fun TabImage(
+    tab: TabSessionState,
+    modifier: Modifier = Modifier,
+    thumbnailDescription: String? = null,
+    contentScale: ContentScale = ContentScale.FillWidth,
+    alignment: Alignment = Alignment.TopCenter
+) {
+    val previewImageUrl = tab.content.previewImageUrl
+    val thumbnail = tab.content.thumbnail
+
+    when {
+        !previewImageUrl.isNullOrEmpty() -> {
```

Comment:
I don't think you can refactor this for the tabs tray because it will always prioritize displaying the hero image, fallback to thumbnail. This image ordering is specific to recent tabs and bookmarks.

Reply:
Is there any specific reason about why we couldn't use the hero image on the tabs tray too?

Reply:
That's a UX question.

Reply:
You would effectively be switching all the thumbnails to be using the hero image with this change assume every site you visit has a hero image. For that reason alone, I don't think we want to pursue this.

Reply:
I update the approach :)